x-pack/plugins/security_solution/public/management/pages/trusted_apps/state/async_data_binding.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Immutable } from '../../../../../common/endpoint/types';
+
+export interface UninitialisedAsyncBinding {
+  type: 'UninitialisedAsyncBinding';
+}
+
+export interface InProgressAsyncBinding<T, E> {
+  type: 'InProgressAsyncBinding';
+  previousBinding: StaleAsyncBinding<T, E>;
+}
+
+export interface PresentAsyncBinding<T> {
+  type: 'PresentAsyncBinding';
+  data: T;
+}
+
+export interface FailedAsyncBinding<T, E> {
+  type: 'FailedAsyncBinding';
+  error: E;
+  lastPresentBinding?: PresentAsyncBinding<T>;
+}
+
+export type StaleAsyncBinding<T, E> =
+  | UninitialisedAsyncBinding
+  | PresentAsyncBinding<T>
+  | FailedAsyncBinding<T, E>;
+
+export type AsyncDataBinding<T, E> =
+  | UninitialisedAsyncBinding
+  | InProgressAsyncBinding<T, E>
+  | PresentAsyncBinding<T>
+  | FailedAsyncBinding<T, E>;
+
+export const isUninitialisedAsyncBinding = <T, E>(
+  binding: Immutable<AsyncDataBinding<T, E>>
+): binding is Immutable<UninitialisedAsyncBinding> => binding.type === 'UninitialisedAsyncBinding';
+
+export const isInProgressBinding = <T, E>(
+  binding: Immutable<AsyncDataBinding<T, E>>
+): binding is Immutable<InProgressAsyncBinding<T, E>> => binding.type === 'InProgressAsyncBinding';
+
+export const isPresentAsyncBinding = <T, E>(
```

Comment:
is this is the current data binding for the list?

Reply:
All types were renamed.